### PR TITLE
[PLAT-7566] Add maze runner tests for network breadcrumbs

### PR DESCRIPTION
**Pull request by gingerbenw**
## Goal

To ensure the network breadcrumb functionality is appropriately tested

### CODE REVIEW

File: test/browser/features/fixtures/network_breadcrumbs/script/fetch.html [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      })
+    
+    </script>
+  </head>
+  <body>
+    <h1>Fetch tests</h1>
+    <pre id="bugsnag-test-should-run">PENDING</pre>
+    <script>
+      var el = document.getElementById('bugsnag-test-should-run')
+      el.textContent = el.innerText = (typeof Promise !== 'undefined' && typeof Promise.all !== 'undefined' && 'fetch' in window)
```
Comment (djskinner) [inline test/browser/features/fixtures/network_breadcrumbs/script/fetch.html:30]: what is `el.textContent = el.innerText` doing out of interest? (I haven't reviewed many maze runner test PRs)

File: test/browser/features/network_breadcrumbs.feature [hunk 2]
```diff
@@ -0,0 +1,12 @@
+@network_breadcrumbs
+Feature: Network breadcrumbs
+
+Scenario: fetch
+  When I navigate to the test URL "/network_breadcrumbs/script/fetch.html"
+  And I wait to receive an error
+  Then the error is a valid browser payload for the error reporting API
```
Comment (djskinner) [inline test/browser/features/network_breadcrumbs.feature:7]: should there be some way to assert that the "valid browser payload" actually contained a network breadcrumb or two? Or is that the kind of thing we do at unit test level only?

**Review by djskinner — APPROVED** [review]
sftm. just a couple of questions relating to my lack of experience with maze runner

File: test/browser/features/fixtures/network_breadcrumbs/script/fetch.html (hunk 1 above)
Comment (gingerbenw) [reply test/browser/features/fixtures/network_breadcrumbs/script/fetch.html:30]: I think we're assigning Yes/No to both `textContent` and `innerText` to support multiple types of html tags, I'm confident that `textContent` would cover all situations though, I was simply following suit!

File: test/browser/features/network_breadcrumbs.feature (hunk 2 above)
Comment (gingerbenw) [reply test/browser/features/network_breadcrumbs.feature]: I've added some tests to validate a successful and failure for both fetch and xmlhttprequest breadcrumbs 👍🏻

File: test/browser/features/network_breadcrumbs.feature [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    # Successful request
+    And the event "breadcrumbs.4.type" equals "request"
+    And the event "breadcrumbs.4.name" equals "fetch() succeeded"
+    And the event "breadcrumbs.4.timestamp" is a timestamp
+    And the event "breadcrumbs.4.metaData.status" equals 200
+    # Unsuccessful request
+    And the event "breadcrumbs.5.type" equals "request"
+    And the event "breadcrumbs.5.name" equals "fetch() failed"
+    And the event "breadcrumbs.5.timestamp" is a timestamp
+    And the event "breadcrumbs.5.metaData.status" equals 404
```
Comment (djskinner) [inline test/browser/features/network_breadcrumbs.feature:17]: 👍

**Comment by gingerbenw** [comment]
Sorry @djskinner - a good few more changes since last time, could you re-review please and thankyou?

File: test/browser/features/support/env.rb [hunk 4]
```diff
@@ -28,6 +28,16 @@ def get_error_message id
...
+Before('@skip_ie_8') do |scenario|
+  browser = Maze.config.browser
+  skip_this_scenario unless browser != "ie_8"
+end
+
+Before('@skip_ie_9') do |scenario|
+  browser = Maze.config.browser
+  skip_this_scenario unless browser != "ie_9"
+end
+
```
Comment (gingerbenw) [inline test/browser/features/support/env.rb:40]: @joshedney just wanted to check you're happy with these statements

Comment (joshedney) [reply test/browser/features/support/env.rb:40]: LGTM!

**Review by joshedney — APPROVED** [review]
Approving MR Changes

**Pull request merged**